SoundEngine/famistudio_cc65.h

```diff
@@ -0,0 +1,126 @@
+ * ======================================================================================================================
+ */
+
+void __fastcall__ famistudio_music_play(unsigned char song_index);
+
+/**
+ * ======================================================================================================================
+ * famistudio_music_pause (public)
+ * 
+ * Pause/unpause the currently playing song. Note that this will not update the APU, so sound might linger. Calling
+ * famistudio_update after this will update the APU.
+ * 
+ * [in] mode : zero to play, non-zero to pause.
+ * ======================================================================================================================
+ */
+#define FAMISTUDIO_PLAY 0
+#define FAMISTUDIO_PAUSE 1
+
+void __fastcall__ famistudio_music_pause(unsigned char mode);
+
+/**
+ * ======================================================================================================================
+ * famistudio_music_stop (public)
+ * 
+ * Stops any music currently playing, if any. Note that this will not update the APU, so sound might linger. Calling
+ * famistudio_update after this will update the APU.
+ * 
+ * [in] no input params.
+ * ======================================================================================================================
+ */
+
+void __fastcall__ famistudio_music_stop(void);
+
+/**
+ * ======================================================================================================================
+ * famistudio_update (public)
+ * 
+ * Main update function, should be called once per frame, ideally at the end of NMI. Will update the tempo, advance
+ * the song if needed, update instrument and apply any change to the APU registers.
+ * 
+ * [in] no input params.
+ * ======================================================================================================================
+ */
+
+void __fastcall__ famistudio_update(void);
+
+#ifdef FAMISTUDIO_CFG_SFX_SUPPORT
+
+/**
+ * ======================================================================================================================
+ * famistudio_sfx_init(public)
+ * 
+ * Initialize the sound effect player.
+ * 
+ * [in] sfx_data: Sound effect data pointer
+ * ======================================================================================================================
+ */
+
+void __fastcall__ famistudio_sfx_init(void* sfx_data);
+
+
+/**
+ * ======================================================================================================================
+ * famistudio_sfx_play (public)
+ * 
+ * Plays a sound effect.
+ * 
+ * [in] sfx_index: Sound effect index (0...127)
+ * [in] channel: Offset of sound effect channel, should be FAMISTUDIO_SFX_CH0..FAMISTUDIO_SFX_CH3
+ * ======================================================================================================================
+ */
+
+// #define FAMISTUDIO_SFX_STRUCT_SIZE = 15
+#define FAMISTUDIO_SFX_CH0 0
```

Comment:
Annoyingly I couldn't find anyway to load or import a constant from ASM into C, so I had to redefine this. I used the same name to try and help with Ctrl+F when refactoring. But this is probably something that'll come to haunt you if you change the SFX stuff :(